extmod/network_ninaw10.c

```diff
@@ -536,17 +536,31 @@ STATIC int network_ninaw10_socket_setsockopt(mod_network_socket_obj_t *socket, m
 }
 
 STATIC int network_ninaw10_socket_settimeout(mod_network_socket_obj_t *socket, mp_uint_t timeout_ms, int *_errno) {
-    if (timeout_ms == UINT32_MAX) {
-        // no timeout is given, set the socket to blocking mode.
-        timeout_ms = 0;
-    }
     socket->timeout = timeout_ms;
     return 0;
 }
 
 STATIC int network_ninaw10_socket_ioctl(mod_network_socket_obj_t *socket, mp_uint_t request, mp_uint_t arg, int *_errno) {
-    *_errno = MP_EIO;
-    return -1;
+    mp_uint_t ret = 0;
+    if (request == MP_STREAM_POLL) {
+        if (arg & MP_STREAM_POLL_RD) {
+            uint16_t avail = 0;
+            if (nina_socket_avail(socket->fileno, socket->type, &avail) != 0) {
```

Comment:
I'm not sure `socket->type` has the correct value to pass to `nina_socket_avail`, it needs to convert, eg, `MOD_NETWORK_SOCK_STREAM` to `NINA_SOCKET_TYPE_TCP`.

Reply:
Good find! It doesn't cause an issue but I fixed it anyway.